apis/management.cattle.io/v3/rke_types.go

```diff
@@ -23,6 +23,8 @@ type RancherKubernetesEngineConfig struct {
 	Authorization AuthzConfig `yaml:"authorization" json:"authorization,omitempty"`
 	// Enable/disable strict docker version checking
 	IgnoreDockerVersion bool `yaml:"ignore_docker_version" json:"ignoreDockerVersion" norman:"default=true"`
+	// Is this a DinD install?
+	DinD bool `yaml:"dind,omitempty" json:"dind,omitempty"`
```

Comment:
This is really a part of #648 and related to https://github.com/rancher/rke/pull/1056

Reply:
Based on the offline talk with @jgreat and @galal-hussein, we are going to support dind flag in rke internal cluster structure only, as it will be set using cli only.